src/script/standard.cpp

```diff
@@ -152,7 +152,6 @@ txnouttype Solver(const CScript& scriptPubKey, std::vector<std::vector<unsigned
         return TX_MULTISIG;
     }
 
-    vSolutionsRet.clear();
```

Comment:
The `clear` here is for documentation purposes (and clearing an empty vector is free anyway)

Reply:
If it's documentation then maybe change to `assert(vSolutionsRet.empty())`?

Reply:
Would suggest keeping the line - it enforces an invariant against future change, and would be good to consider it separately if we are to change it. `assert` could be overkill given this is basically a failure case anyway.

Reply:
yeah but assert would catch a situation where you missed a return higher up which is something that would definitely be good to catch right away.

Reply:
Now using `assert(...)`. It's an unrelated change and I don't mind dropping it but either way `assert()` looks better to me.